### PR TITLE
GMT CG Hub - Add command to help TCGA imports

### PR DESCRIPTION
**Pull request by ebelter**

### CODE REVIEW

**Comment by apregier** [comment]
+1
It did take me a long time to understand this code, but I guess a lot of it is cleaning up the confusing data structure that you get back from CgHub

File: lib/perl/Genome/Model/Tools/CgHub/Metadata.pm [hunk 1]
```diff
@@ -13,175 +13,143 @@ class Genome::Model::Tools::CgHub::Metadata {
...
-    else {
-        # unknown library strategy
-        $target_region = 'none';
-    }
+    my $result = $self->metadata->{$lookup_id};
+    return if not $result;
+    my $value = $result->{$name};
 
-    return $target_region;
+    return if not defined $value or $value eq '';
```
Comment (apregier) [inline lib/perl/Genome/Model/Tools/CgHub/Metadata.pm]: This line is pretty redundant with the next line, right?

Comment (ebelter) [inline lib/perl/Genome/Model/Tools/CgHub/Metadata.pm]: Yeah - There is some code leftover as I updating having the metadata hold multiple analysis ids.

File: lib/perl/Genome/Model/Tools/CgHub/Metadata.pm [hunk 2]
```diff
@@ -13,175 +13,143 @@ class Genome::Model::Tools::CgHub::Metadata {
...
+    return $value;
 }
 
 sub _files {
-    my $self = shift;
+    my ($self, $lookup_id) = Params::Validate::validate_pos(@_, {type => HASHREF}, {type => SCALAR});
```
Comment (tmooney) [inline lib/perl/Genome/Model/Tools/CgHub/Metadata.pm]: Some places are validating `$self` as an `OBJECT` while others are validating it as a `HASHREF`.  They should probably all be `OBJECT` (or alternatively `{isa => __PACKAGE__}` or the like).

**Pull request merged**